static/app/utils/performanceForSentry.tsx

```diff
@@ -27,3 +28,44 @@ export function onRenderCallback(
     // Add defensive catch since this wraps all of App
   }
 }
+
+export const VisuallyCompleteWithData = ({
+  id,
+  hasData,
+  children,
+}: {
+  children: ReactNode;
+  hasData: boolean;
+  id: string;
+}) => {
+  const isVisuallyCompleteSet = useRef(false);
+  const isDataCompleteSet = useRef(false);
+  useEffect(() => {
+    try {
+      const transaction: any = getCurrentSentryReactTransaction(); // Using any to override types for private api.
+      const now = performance.now();
+      if (!isVisuallyCompleteSet.current) {
+        transaction.setMeasurements({
+          ...transaction._measurements,
+          visuallyComplete: {value: now},
+        });
+        isVisuallyCompleteSet.current = true;
+      }
+      if (!isDataCompleteSet.current) {
+        transaction.setMeasurements({
+          ...transaction._measurements,
+          visuallyCompleteData: {value: now},
+        });
+        isDataCompleteSet.current = true;
+      }
+    } catch (_) {
+      // Defensive catch since this code is auxiliary.
```

Comment:
we should log to sentry to check if it's working

Reply:
The existence of the measurement is sufficient, don't want to cause extra load on this since it's unknown if it'll work.